diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt

```diff
@@ -155,26 +153,43 @@ class BlockStructureBraces : Rule("block-structure") {
             CLASS, OBJECT_DECLARATION -> node.findChildByType(CLASS_BODY)!!.findChildAfter(LBRACE, WHITE_SPACE)
             else -> node.findChildByType(BLOCK)?.findChildAfter(LBRACE, WHITE_SPACE)
         }
-        if (nodeText != null && !nodeText.text.contains("\n".toRegex())) {
-            BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "incorrect line after opening brace",
-                    nodeText.startOffset) {
+        if (newNode != null && !newNode.text.contains("\n")) {
+            BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "incorrect same line after opening brace",
+                    newNode.startOffset) {
+            }
+        }
+    }
+
+    private fun checkMidBrace(node: ASTNode){
+        val allMiddleSpace = when(node.elementType){
+            TRY -> {
+                node.findAllNodesWithSpecificType(FINALLY).map { it.prevSibling { type -> type.elementType == WHITE_SPACE } } +
+                node.findAllNodesWithSpecificType(CATCH).map { it.prevSibling { type -> type.elementType == WHITE_SPACE } }
+            }
+            else -> node.findAllNodesWithSpecificType(THEN).map { it.nextSibling { type -> type.elementType == WHITE_SPACE } }
+        }
+        allMiddleSpace.forEach {
+            if (it!!.text.contains("\n")){
+                BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "incorrect new line after closing brace",
+                        it.startOffset) {
+                }
             }
         }
     }
 
-    private fun checkCloseBrace(node: ASTNode?, configuration: BlockStructureBracesConfiguration) {
+    private fun checkCloseBrace(node: ASTNode, configuration: BlockStructureBracesConfiguration) {
         if (!configuration.closeBrace) return
-        val space = node?.getChildren(null)?.findLast { it.elementType == WHITE_SPACE }
-        if (space != null && !space.text.contains("\n".toRegex())) {
-            BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "incorrect line after closing brace",
-                    (space).startOffset) {
+        val space = node.getChildren(TokenSet.WHITE_SPACE).lastOrNull()
```

Comment:
Also I think it would be better to look for white space before last `LBRACE`. There could be no white space between code and closing brace, and you will get wrong node here.